napari/components/dims.py

```diff
@@ -358,7 +359,7 @@ def set_axis_label(
         full_axis_labels = list(self.axis_labels)
         for ax, val in zip(axis, label):
             full_axis_labels[ax] = val
-        self.axis_labels = full_axis_labels
+        self.axis_labels = tuple(full_axis_labels)
```

Comment:
Comment: this is handled by the validator: https://github.com/napari/napari/blob/864031375b825951a39348b9ba55b48b10fc395a/napari/components/dims.py#L112

But I guess mypy has no way to infer that.

Reply:
I'm not familiar with validators, but I guess they're a runtime check instead of a static markup, so `mypy` doesn't know about them.